File: leetcode/Tow Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> mymap = new HashMap<> ();
        for(int i = 0; i < nums.length; i++)
        {
            int x = target - nums[i];
            if(mymap.get(x) != null)
            {
                return new int[] {mymap.get(x), i};
            }
            else
            {
                mymap.put(nums[i], i);
            }
        }
        return new int[] {-1, -1};
    }
}

File: Leetcode/CPP/Tow Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mymap;
        
        for(int i = 0; i < nums.size(); i++)
        {
            int x = target - nums[i];
            if (mymap.find(x) != mymap.end())
            {
                return {mymap[x], i};
            }
            mymap[nums[i]] = i;
        }
        return {};
    }
};
